hooks/usePaprBalance/usePaprBalance.tsx

```diff
@@ -29,19 +29,13 @@ export function PaprBalanceProvider({ children }: PropsWithChildren<{}>) {
     undefined,
   );
 
-  const tokenAddressForBalanceRead = useMemo(() => {
```

Comment:
i think because this provider is SSRed, we can't exactly memo this -- still unsure why this fixes things

Reply:
Can you add a bit of detail as to what the problem is? Not sure I understand

Reply:
this actually makes sense now -- memoized value is SSRed and doesn't change on load, hook is generated at run time on client probably so passing values directly works